app/controllers/github_sync_controller.rb

```diff
@@ -9,7 +15,11 @@ def sync
     repos = github_sync.run
 
     if contains_invalid_repository? repos
-      @repos = Github.repos
+      github = Github.new
+      repos = github.get_all_repos
```

Comment:
I believe in this line the variable 'repos' defined earlier is overshadowed; as a result line 23 does not work properly, because 'repos' no longer holds an array of AR objects. So in case there is an error when saving projects, the red message shows up sayng 'there were problems creating', but the per-project error messages are not displayed.

It seems that there might be some duplication between this action and the index action - perhaps possible to extract into a method.

Also, if you go to e.g. 2nd page of projects, select a repo that has errors, you get redirected with the error message to the 1st page of repos rather than 2nd.